src/sentry/grouping/ingest/seer.py

```diff
@@ -176,9 +178,26 @@ def _circuit_breaker_broken(event: Event, project: Project) -> bool:
     return circuit_broken
 
 
+def _has_empty_stacktrace_string(event: Event, variants: Mapping[str, BaseVariant]) -> bool:
+    stacktrace_string = get_stacktrace_string(get_grouping_info_from_variants(variants))
+    # store the stacktrace string in the event so we only calculate it once and so that
+    # we only calculate it if none of the previous checks in should_call_seer_for_grouping fail
+    event.data["stacktrace_string"] = stacktrace_string
```

Comment:
Should we wait to store the stacktrace string on the event until after we know it's valid, i.e., between `return True` and `return False`?

Reply:
yup! good callout